File: src/main/java/atm/bloodworkxgaming/oeintegration/ModItems.java
package atm.bloodworkxgaming.oeintegration;

import atm.bloodworkxgaming.oeintegration.Common.Items.ItemExcavateModifier;
import net.minecraft.item.Item;
import net.minecraftforge.fml.relauncher.Side;
import net.minecraftforge.fml.relauncher.SideOnly;
import net.minecraftforge.registries.IForgeRegistry;

/**
 * Created by Jonas on 09.03.2017.
 */
public class ModItems {

    public static ItemExcavateModifier itemExcavateModifier;

    static {
        itemExcavateModifier = new ItemExcavateModifier();
    }

    public static void registerItems(IForgeRegistry<Item> registry) {
        registry.register(itemExcavateModifier);
    }


    @SideOnly(Side.CLIENT)
    public static void initModels() {
        itemExcavateModifier.initModel();
    }
}

File: src/main/java/atm/bloodworkxgaming/oeintegration/ModItems.kt
package atm.bloodworkxgaming.oeintegration

import net.minecraft.item.Item
import net.minecraft.item.ItemGroup
import net.minecraftforge.registries.ForgeRegistries
import thedarkcolour.kotlinforforge.forge.KDeferredRegister

/**
 * Created by Jonas on 09.03.2017.
 */
object ModItems {
    val REGISTRY = KDeferredRegister(ForgeRegistries.ITEMS, OreExcavationIntegration.MOD_ID)

    val ITEM_EXCAVATE_MODIFIER by REGISTRY.registerObject("excavatemodifier") {
        println("should register item now")
        Item(Item.Properties().tab(ItemGroup.TAB_TOOLS))
    }

    /*@SideOnly(Side.CLIENT)
    fun initModels() {
        itemExcavateModifier.initModel()
    }

    init {
        itemExcavateModifier = ItemExcavateModifier()
    }*/

}
